featuretools/entityset/entityset.py

```diff
@@ -852,58 +853,58 @@ def normalize_dataframe(self, base_dataframe_name, new_dataframe_name, index,
-    #         combined_df = pd.concat([self_df, other_df])
-    #         if entity.created_index == entity.index:
-    #             columns = [col for col in combined_df.columns if
-    #                        col != entity.index or col != entity.time_index]
-    #         else:
-    #             columns = [entity.index]
-    #         combined_df.drop_duplicates(columns, inplace=True)
-
-    #         if entity.time_index:
-    #             combined_df.sort_values([entity.time_index, entity.index], inplace=True)
-    #         else:
-    #             combined_df.sort_index(inplace=True)
-    #         if (entity.last_time_index is not None or
-    #                 other[entity.id].last_time_index is not None):
-    #             has_last_time_index.append(entity.id)
-
-    #         combined_es.update_dataframe(
-    #             entity_id=entity.id,
-    #             df=combined_df,
-    #             recalculate_last_time_indexes=False,
-    #         )
-
-    #     combined_es.add_last_time_indexes(updated_dataframes=has_last_time_index)
-    #     self.reset_data_description()
-    #     return combined_es
+    def concat(self, other, inplace=False):
+        '''Combine entityset with another to create a new entityset with the
+        combined data of both entitysets.
+        '''
+        if not self.__eq__(other):
+            raise ValueError("Entitysets must have the same dataframes, relationships"
+                             ", and column names")
+
+        if inplace:
+            combined_es = self
+        else:
+            combined_es = copy.deepcopy(self)
+
+        lib = pd
+        if ks and isinstance(self.dataframes[0], ks.DataFrame):
+            lib = ks
+        elif dd and isinstance(self.dataframes[0], dd.DataFrame):
+            lib = dd
+
+        has_last_time_index = []
+        for df in self.dataframes:
+            self_df = df
+            other_df = other[df.ww.name]
+            combined_df = lib.concat([self_df, other_df])
+            # If both DataFrames have made indexes, there will likely
+            # be overlap in the index column, so we use the other values
+            if self_df.ww.make_index or other_df.ww.make_index:
+                columns = [col for col in combined_df.columns if
+                           col != df.ww.index or col != df.ww.time_index]
+            else:
+                columns = [df.ww.index]
+            combined_df.drop_duplicates(columns, inplace=True)
+
+            self_lti_col = df.ww.metadata.get('last_time_index')
+            other_lti_col = other[df.ww.name].ww.metadata.get('last_time_index')
+            if (self_lti_col is not None or
+                    other_lti_col is not None):
+                has_last_time_index.append(df.ww.name)
+
+            combined_es.update_dataframe(
+                dataframe_name=df.ww.name,
+                df=combined_df,
+                recalculate_last_time_indexes=False,
+                already_sorted=False
```

Comment:
The previous implementation sorted on just the index column even when there was no time index. `EntitySet.update_dataframe` already sorts on time index and index if there's a time index.

If there's a reason to just sort on the index after the `concat` call, happy to add it back in, but we only ever sort when there's a time index, so I just stuck with the `update_dataframe` sorting for now.